File: readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

void readability(char *str)
{
    int i = 0;
    int letter = 0;
    int word = 1;
    int sentences = 0;
    while (str[i] != 0)
    {
        if ((str[i] > 64 && str[i] < 91) || (str[i] > 96 && str[i] < 123))
            letter++;
        if (str[i] == 32)
            word++;
        if (str[i] == 63 || str[i] == 46 || str[i] == 21)
            sentences++;
        i++;
    }
    float averageWords = (100 / (float) word) * (float) letter;
    float averageSentences = (100 / (float) word) * (float) sentences;
    int index = round(0.0588 * averageWords - 0.296 * averageSentences - 15.8);

    if (index < 1)
        printf("Before Grade 1\n");
    else if (index > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %d\n", index);
}

int main(void)
{
    string message = get_string("Text: ");
    readability(message);
}

File: sentimental-readability/readability.py
from cs50 import get_string


def readability(message):
    char = 0
    index = 0
    letter = 0
    word = 1
    sentences = 0
    for char in message:
        index += 1
        if char.isalpha():
            letter += 1
        elif char.isspace():
            word += 1
        elif char == "." or char == "!" or char == "?":
            sentences += 1
    averageWords = (100 / word) * letter
    averageSentences = (100 / word) * sentences
    index = round(0.0588 * averageWords - 0.296 * averageSentences - 15.8)
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


message = get_string("Text: ")
readability(message)
